runtime/src/anchor.rs

```diff
@@ -283,6 +336,26 @@ impl<T: Trait> Module<T> {
         }
     }
 
+    /// iterate from the last evicted anchor to latest anchor, while removing indexes that
+    /// are no longer valid because they belong to an expired/evicted anchor. The loop is
+    /// only allowed to run MAX_LOOP_IN_TX at a time.
+    fn remove_anchor_indexes(yesterday: u32) -> usize {
+        (LatestEvictedAnchorIndex::get() + 1..LatestAnchorIndex::get() + 1)
+            // limit to only MAX_LOOP_IN_TX number of anchor indexes to remove
+            .take(MAX_LOOP_IN_TX as usize)
+            // get eviction date of the anchor given by index
+            .map(|idx| {(idx, <AnchorEvictDates<T>>::get(<AnchorIndexes<T>>::get(idx)))})
+            // filter out evictable anchors, anchor_evict_date can be 0 when evicting before any anchors are created
+            .filter(|(_, anchor_evict_date)| anchor_evict_date <= &yesterday && anchor_evict_date > &0)
```

Comment:
what happens when the first 500 all are continuously having `anchor_evict_date=0` is that a possible scenario? could that happen and mean that each run of this evict function it would essentially pick up 0 anchors to actually evict?

Reply:
is there a test-case that can be written for this?

Reply:
remove > 0 check